File: JavaScript/6js 2/uniqueInOrder.js
/*
|--------------------------------------------------------------------------
| Unique In Order
|--------------------------------------------------------------------------
|
*/

function uniqueInOrder(iterable) {
   if (!iterable) return [];
   let res = [iterable[0]];
   for (let i = 1; i < iterable.length; i++) {
      if (iterable[i] !== iterable[i - 1]) {
         res.push(iterable[i]);
      }
   }
   return res;
}


var uniqueInOrder = function (iterable) {
   return [...iterable].filter((a, i) => a !== iterable[i - 1]);
};


var uniqueInOrder = function (iterable) {
   var res = [];
   for (var i = 0; i < iterable.length; i++) {
      if (iterable[i] != iterable[i + 1]) res.push(iterable[i]);
   }
   return res;
};


console.log(uniqueInOrder("AAAABBBCCDAABBB"), ["A", "B", "C", "D", "A", "B"]);

File: Python/6py 2/unique_in_order.py
"""
|--------------------------------------------------------------------------
| Unique In Order
|--------------------------------------------------------------------------
|
"""

def unique_in_order(sequence):
   if not sequence: return []
   res = [sequence[0]]
   for i in range(1, len(sequence)):
      if sequence[i] != sequence[i - 1]:
         res.append(sequence[i])
   return res


unique_in_order = lambda l: [z for i, z in enumerate(l) if i == 0 or l[i - 1] != z]


from itertools import groupby
def unique_in_order(iterable):
   return [k for (k, _) in groupby(iterable)]


print(unique_in_order("AAAABBBCCDAABBB"), ["A", "B", "C", "D", "A", "B"]);